server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts

```diff
@@ -591,19 +607,24 @@ export class SummaryWriter implements ISummaryWriter {
                     $lt: lt,
                 },
             };
-            const logTail = await this.opStorage.find(query, { "operation.sequenceNumber": 1 });
-
-            // If the db is not updated with all logs yet, get them from checkpoint messages.
-            if (logTail.length !== (lt - gt - 1)) {
-                const nextSeq = logTail.length === 0 ? gt : logTail[logTail.length - 1].operation.sequenceNumber + 1;
-                for (const message of pending) {
-                    if (message.operation.sequenceNumber >= nextSeq && message.operation.sequenceNumber < lt) {
-                        logTail.push(message);
-                    }
+
+            const logTailOpMessage = await this.opStorage.find(query, { "operation.sequenceNumber": 1 });
+            logTail = logTailOpMessage.map((log) => log.operation);
+        }
+        
+        Lumberjack.info(`LogTail of length ${logTail.length} fetched from seq no ${gt} to ${lt}`, this.lumberProperties);
+
+        // If the db is not updated with all logs yet, get them from checkpoint messages.
+        if (logTail.length !== (lt - gt - 1)) {
+            Lumberjack.info(`Populating logtail gaps`, this.lumberProperties);
+            const nextSeq = logTail.length === 0 ? gt : logTail[logTail.length - 1].sequenceNumber + 1;
```

Comment:
>logTail[logTail.length - 1].sequenceNumber + 1 [](http://example.com/codeflow?start=56&length=46)

Is it guaranteed that we'd only miss trailing ops?

Reply:
This part of the code was not changed as a part of this PR but my understanding of this code is that we could only miss the trailing ops. Will be sure to add logs to catch the other scenarios if they happen